src/agents/workflows/subgraphs/step2_router/graph.py

```
@@ -12,21 +12,71 @@
+                    "Question:\n"
+                    f"{latest_question}\n\n"
+                    "Answer with either STEM or NON_STEM."
+                )
+            ),
+        ]
+        try:
+            ai_message = classifier.invoke(prompt_messages)
+            verdict = getattr(ai_message, "content", "")
+            normalized = (verdict or "").strip().upper()
+            is_stem = normalized.startswith("STEM")
+        except Exception as exc:  # pragma: no cover - defensive logging
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)